Android/app/src/main/java/com/example/boardgamesocial/MainApp/Fragments/SingleHostedGameFragment.java

```diff
@@ -98,9 +124,261 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+
+            availableSeats = processJoinGame(availableSeats, updatedHostedGame);
+
+            btnJoinGame.setText(doesHostedGameToUserExists(updatedHostedGame.getGameId()) ? "Un-Join Game" : "Join Game");
+
+            tvSeatsAvailable.setText(String.format(getString(R.string.game_seats_available_text), availableSeats));
+
+        });
+    }
+
+    private HostedGame getHostedGameByGameId(Integer eventId, Integer gameId) {
+
+        AtomicReference<HostedGame> hostedGame = new AtomicReference<>();
+
+        try {
+
+            retrofitClient.getCall(HostedGame.class, new HashMap<String, String>() {{
+                put("eventId", String.valueOf(eventId));
+                put("gameId", String.valueOf(gameId));
+            }}).blockingSubscribe(hostedGameJson -> {
+                List<HostedGame> hostedGameList = getObjectList(hostedGameJson, HostedGame.class);
+                if (!hostedGameList.isEmpty()) {
+                    hostedGame.set(hostedGameList.get(0));
+                }
+            });
+
+        } catch(Exception ex) {
+            ex.printStackTrace();
+        }
+
+        return hostedGame.get();
+    }
+
+    private int processJoinGame(int seatsAvailable, HostedGame hostedGameToJoin) {
+
+        Log.i(TAG, "Inside processJoinEvent with seatsAvailable: " + seatsAvailable);
+
+        try {
+
+            if(insertHostedGameToUser(hostedGameToJoin.getGameId())) {
+                insertEventToUser(hostedGameToJoin.getEventId());
+                seatsAvailable--;
+            } else {
+                seatsAvailable++;
+                deleteHostedGameToUser(hostedGameToJoin);
+            }
+
+            hostedGameToJoin.setSeatsAvailable(seatsAvailable);
+            updateHostedGameAvailableSeats(hostedGameToJoin);
+
+        } catch(Exception ex) {
+            ex.printStackTrace();
+        }
+
+        return seatsAvailable;
+    }
+
+    private void deleteHostedGameToUser(HostedGame hostedGame) {
+
+        try {
+
+            if(doesHostedGameToUserExists(hostedGame.getGameId())) {
+                int userId = Utils.getUserId();
+                retrofitClient.deleteCall(HostedGameToUser.class, new HashMap<String, String>() {{
+                    put("gameId", String.valueOf(hostedGame.getGameId()));
+                    put("userId", String.valueOf(userId));
+                }}).blockingSubscribe(deletedHostedGameToUserJson -> {
+                    retrofitClient.getCall(HostedGameToUser.class, new HashMap<String, String>() {{
+                        put("userId", String.valueOf(userId));
+                    }}).blockingSubscribe(hostedGameToUserJson -> {
+                        List<HostedGameToUser> hostedGameToUserList = getObjectList(hostedGameToUserJson, HostedGameToUser.class);
+                        if (hostedGameToUserList.isEmpty()) {
+                            deleteEventToUser(hostedGame.getEventId());
+                        }
+                    });
+                });
```

Comment:
It seems you are not using response from deleteCall, in that case you can tring using doOnComplete, since it will not take in emitted data, but subscribers should work too.

Reply:
Ok, again will have to take a look and add it in a pr called fixes